### ISSUE TITLE
NotifyIcon balloon is hided when FormBorderStyle property is set to None and form is maximized

### ISSUE THREAD

**Issue by Marie-Shi** [dotnet/winforms#4315]
* .NET Core Version: 
.Net 6.0.100-alpha.1.20602.2 from master branch

* Have you experienced this same bug with .NET Framework?: 
Yes

**More info:**
This is not a regression issue and it can also repro on Framework project.

**Minimal repro:**
1.	Extract “**NotifyIcon_hide_issue.zip**” and open it in VS.
[NotifyIcon_hide_issue.zip](https://github.com/dotnet/winforms/files/5633442/NotifyIcon_hide_issue.zip)

2.	Build and run this project.
3.	Click ‘**Border None**’ button first, next click ‘**Change State**’ button to make form maximized.
4.	Click ‘**Notify**’ button and observe whether NotifyIcon balloon is shown.

**Actual behavior:**
NotifyIcon balloon text is not shown on screen(not shown in front of the maximized form), but actually it is hided in Windows Task bar. Please see the attachment:
[NotifyIcon_Hide_video.zip](https://github.com/dotnet/winforms/files/5633440/NotifyIcon_Hide_video.zip)
![Notifyicon_not_show](https://user-images.githubusercontent.com/45864017/100959528-78347f80-3559-11eb-960e-f896506645ff.png)

**Expected behavior:**
The NotifyIcon balloon text should be shown in screen(in front of the maximized form) when the form is maximized.
![notifyicon_show](https://user-images.githubusercontent.com/45864017/100959500-5fc46500-3559-11eb-8977-6fe61bdeb5f0.png)

**Comment by merriemcgaw** [comment]
This makes sense as the expected behavior. Think of a point of sale app/kiosk app with no border - you wouldn't want any notifications popping up when their main app is running. 

Resolving by design.

**Comment by weltkante** [comment]
Just for the record, if you cover a whole monitor (including the taskbar) then the shell will assume you are a full screen application and move itself out of the way. Depending on your Windows settings this will mean notifications are suspended, but you can configure Windows to deliver notifications while a full screen application runs.

For example I have 3 monitors so I configured Windows this way, its silly not getting notifications when one monitor runs a fullscreen application, so I don't like that default. With my setup I could not reproduce the behavior you were seeing.

So in summary, the behavior you are seeing is Windows treating you as a fullscreen application, and there are user settings to tweak the notification behavior in presence of a fullscreen application. This is nothing your app should be concerned about, its the users choice how he wants to configure this. There certainly is nothing WinForms should be doing about it.

If you don't want to be treated as a fullscreen application you should not cover the taskbar (use e.g. `Screen.GetWorkingArea` to determine the area available for windowed applications)
